### DISCUSSION TITLE
Can TRPC be used without separate backend?

### DISCUSSION THREAD

**Discussion by hanshank** [trpc/trpc#4487] [General]
Is there a way to use TRPC without defining a backend route for TRPC? My team have been trying out TRPC using the api endpoints in Next.js. We have some temporary restrictions where we need to make certain upstream API calls client-side. Is there a way to call the procedures in a client-side context? And if so, would be awesome if we can still use the hooks inside our React  components for mutations and queries.

Thanks!

**Comment by KATT** [comment]
You could make an `ssrLink`, see #3768
